File: tdd-workflow-example-python/test/greeter_test.py
# -*- coding: utf-8 -*-

import unittest
from ddt import ddt, data, unpack
from app import greeter


@ddt
class GreeterTestCase(unittest.TestCase):
    @unpack
    @data({'expected': 'Hello, World!', 'given': 'World'},
          {'expected': 'Hello, Tests!', 'given': 'Tests'})
    def test_greet(self, expected, given):
        actual = greeter.greet(given)
        self.assertEqual(expected, actual)

    def test_loud_greet(self):
        actual = greeter.loud_greet('')
        self.assertEqual('HI!', actual)

File: tdd-workflow-example-golang/greeting/greeter_test.go
package greeting

import "testing"

func TestGreet(t *testing.T) {
	cases := []struct {
		expected string
		given    string
	}{
		{"Hello, World!", "World"},
		{"Hello, Tests!", "Tests"},
		{"Hi!", ""},
	}

	for _, c := range cases {
		actual := Greet(c.given)
		if c.expected != actual {
			t.Errorf("Greet(%v) = %v, expected %v", c.given, actual, c.expected)
		}
	}
}

func TestLoudGreet(t *testing.T) {
	cases := []struct {
		expected string
		given    string
	}{
		{"HI!", ""},
	}

	for _, c := range cases {
		actual := LoudGreet(c.given)
		if c.expected != actual {
			t.Errorf("LoudGreet(%v) = %v, expected %v", c.given, actual, c.expected)
		}
	}
}
